src/ws/session.c

```diff
@@ -559,6 +559,7 @@ open_session (pam_handle_t *pamh)
       debug ("opening pam session for %s", name);
 
       pam_putenv (pamh, "XDG_SESSION_CLASS=user");
+      pam_putenv (pamh, "XDG_SESSION_TYPE=web");
```

Comment:
What implications does this have? The [manpage](https://www.freedesktop.org/software/systemd/man/pam_systemd.html) or a [Google search](https://www.google.com/search?q=XDG_SESSION_TYPE) seems to indicate that this is pretty much advisory, and not much software is looking at it.

Can this please get a unit or integration test to make sure that it gets propagated all the way through? I. e. with `loginctl show-session $XDG_SESSION_ID`

Reply:
> What implications does this have? The manpage or a Google search seems to indicate that this is pretty much advisory, and not much software is looking at it.

I can't add anything to this.  Systemd has added the type for us, so it is only polite that we use it. :-)

Reply:
Ack, that's fine. I'd still like to see an integration test to make sure that this gets propagated all the way to logind.

Reply:
I have added a test.